src/main/java/org/junitpioneer/jupiter/IssueExtension.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionConfigurationException;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.platform.commons.support.AnnotationSupport;
+
+/**
+ * This class implements the functionality for the {@code @Issue} annotation.
+ *
+ * @see Issue
+ */
+class IssueExtension implements BeforeEachCallback {
```

Comment:
I don't think so, because (when I understand it correctly) the `TestWatcher` has a callback for each of the result, but the extension only needs to publish the annotations value.